crates/shared/src/signature_validator/simulation.rs

```diff
@@ -18,18 +19,54 @@ pub struct Validator {
     signatures: contracts::support::Signatures,
     settlement: H160,
     vault_relayer: H160,
+    web3: Web3,
 }
 
 impl Validator {
+    /// The result returned from `isValidSignature` if the signature is correct
+    const IS_VALID_SIGNATURE_MAGIC_BYTES: &'static str = "1626ba7e";
+
     pub fn new(web3: &Web3, settlement: H160, vault_relayer: H160) -> Self {
         let web3 = ethrpc::instrumented::instrument_with_label(web3, "signatureValidation".into());
         Self {
             signatures: contracts::support::Signatures::at(&web3, settlement),
             settlement,
             vault_relayer,
+            web3: web3.clone(),
         }
     }
 
+    /// Simulate isValidSignature for the cases in which the order does not have
+    /// pre-interactions
+    async fn simulate_without_pre_interactions(
+        &self,
+        check: &SignatureCheck,
+    ) -> Result<Simulation, SignatureValidationError> {
+        // We simulate the signature verification from the owner contract's
+        // context. This allows us to simulate the actual `isValidSignature` calls that
+        // would happen as part of a settlement
+        let contract = ERC1271SignatureValidator::at(&self.web3, check.signer);
+        let magic_bytes = contract
+            .methods()
+            .is_valid_signature(Bytes(check.hash), Bytes(check.signature.clone()))
+            .call()
+            .await
+            .map(|value| hex::encode(value.0))
+            .map_err(|e| {
+                tracing::error!(
+                    ?e,
+                    "failed to check EIP-1271 signature with isValidSignature"
+                );
+                e
+            })?;
+
+        if magic_bytes != Self::IS_VALID_SIGNATURE_MAGIC_BYTES {
+            return Err(SignatureValidationError::Invalid);
+        }
+
+        Ok(Simulation { gas_used: 0.into() })
```

Comment:
I think to avoid misuse of the function we should return a `bool` instead of a `Simulation` with an incorrect `gas_used` here.

Reply:
Good point, agree.